qcengine/util.py

```diff
@@ -266,6 +266,44 @@ def popen(args: List[str], append_prefix: bool = False,
             ret["stderr"] = error.decode()
 
 
+@contextmanager
+def environ_context(config: Optional['JobConfig'] = None, env: Optional[Dict[str, str]] = None) -> Dict[str, str]:
```

Comment:
interaction of this and [the `env` in `execute()`](https://github.com/MolSSI/QCEngine/pull/116/files#diff-138f99563678a91c2c82995c8537354cR320) not really explored yet, right?

Reply:
Right, I don't think we can dip to that level yet. The idea here was only to tackle the `np.dot`-like calls with this manager.

I bet we could consider tying the two together eventually, but subprocess becomes very program-dependent.